tools/azsdk-cli/Azure.Sdk.Tools.Cli/Commands/CommandFactory.cs

```diff
@@ -30,14 +30,65 @@ public static RootCommand CreateRootCommand(string[] args, IServiceProvider serv
 
             var toolTypes = SharedOptions.GetFilteredToolTypes(args);
 
-            // walk the tools, register them as subcommands for the root command.
-            foreach (var t in toolTypes)
-            {
-                var tool = (MCPTool)ActivatorUtilities.CreateInstance(serviceProvider, t);
-                rootCommand.AddCommand(tool.GetCommand());
-            }
+            var toolInstances = toolTypes
+                .Select(t => (MCPTool)ActivatorUtilities.CreateInstance(serviceProvider, t))
+                .ToList();
+
+            PopulateToolHierarchy(rootCommand, toolInstances);
 
             return rootCommand;
         }
+
+        private static void PopulateToolHierarchy(RootCommand rootCommand, List<MCPTool> toolList)
```

Comment:
One thing I'm just realizing now is how the options cascade will work. For global options, we just add them as root options to the root command and they can be used in any order. But let's say a command at hierarchy 1 or 2 defines some options and the user calls a hierarchy 3 command. Are there currently positional requirements to the hierarchy 1 options in this scenario, or can they be placed anywhere?

Reply:
I am working through a follow-up commit that allows these groupings to have associated options.

Reply:
Ok this should be g2g now.